tests/System.Text.Encodings.Web.Utf8.Tests/UnescapeUrlPathInPlaceTests.cs

```diff
@@ -0,0 +1,18 @@
+﻿using Xunit;
+
+namespace System.Text.Encodings.Web.Utf8.Tests
+{
+    public class UnescapeUrlPathInPlaceTests : UrlEncoderTests
+    {
+        protected override void TestCore(string raw, string expected)
+        {
+            var input = GetBytes(raw);
+
+            var len = UrlEncoder.UnescapeInPlace(input);
+            Assert.True(len <= input.Length);
+
+            var outputDecoded = Encoding.UTF8.GetString(input.Slice(0, len).ToArray());
```

Comment:
To be more specific, there are valid UTF8 coded byte span rejected by `Utf8String` under the reason "invalid characters".

For example: `%F0%A4%AD%A2` is decoded to `𤭢` in `Encoding.UTF8` but rejected in `Utf8String`.